File: algorithms/Student Attendance Record I/Student Attendance Record I.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        eligible = True
        
        for i in range(0, len(s)-2):
            if s[i:i+3] == "LLL":
                eligible = False
        absent = 0
        for i in range(len(s)):
            if s[i] == "A":
                absent +=1 
                
        if absent>=2:
            eligible = False
            
        return(eligible)

File: scripts/Student Attendance Record I/Student Attendance Record I.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int a=0,l=0;
        for(int i=0;i<s.length();i++){
           if(s[i]=='A'){
               a++;
               l=0;
               if(a>=2) return false;
           }
            else if(s[i]=='L'){
                l++;
                if(l>=3) return false;
            }
            else l=0;
        }
        return true;
    }
};
